src/allDrugs.jl

```diff
@@ -0,0 +1,68 @@
+""" In this file we fit all the drugs att once. """
+
+function getODEparamsAll(p::Array{Float64,1}, concentrations::Array{Float64,2})
+    effects = effects = zeros(9, 8, 4)
+
+    k = 1
+    # Scaled drug effect
+    for i=1:4
+        xx = 1.0 ./ (1.0 .+ (p[k] ./ concentrations[:, i]) .^ p[k+1])
+
+        effects[1, :, i] = p[k+2] .+ (p[k+3] - p[k+2]) .* xx
+        effects[2, :, i] = p[k+4] .+ (p[k+5] - p[k+4]) .* xx
+        effects[3, :, i] = p[k+6] .* xx
+        effects[4, :, i] = p[k+7] .* xx
+        effects[5, :, i] .= p[k+8]
+        k+=9
+    end
+    effects[6, :, :] .= floor(p[37]) #nG1
+    effects[7, :, :] .= floor(p[38]) #nG2
+    effects[8, :, :] .= floor(p[39]) #nD1
+    effects[9, :, :] .= floor(p[40]) #nD2
+
+    return effects
+end
+
+function residHillAll(hillParams::Array{Float64,1}, concentrations::Array{Float64,2}, g1::Array{Float64,3}, g2::Array{Float64,3})
+    res = Atomic{eltype(hillParams)}(0.0)
+    params = getODEparamsAll(hillParams, concentrations)
+
+    # Solve for all drugs
+    @threads for j = 1:4
+        @threads for ii = 1:length(concentrations[:, j])
+            atomic_add!(
+                res,
+                cost(
+                    params[:, ii, j],
+                    g1[:, ii, j],
+                    g2[:, ii, j],
+                    Int(floor(params[6, ii, j])),
+                    Int(floor(params[7, ii, j])),
+                    Int(floor(params[8, ii, j])),
+                    Int(floor(params[9, ii, j])),
+                ),
+            )
+        end
+    end
+
+    return res[]
+end
```

Comment:
Is this the correct way to add threads here? I ran it on the notebook, it works, but it is still very slow. like 8 evaluations in 100 seconds. Is there a way to make it faster? @aarmey

Reply:
You have to initialize IJulia with threads: https://github.com/JuliaLang/IJulia.jl/blob/master/README.md#installing-additional-julia-kernels

Reply:
The best way to speed this up would be to profile the cost function in a unit test. Then, we can look at where the code is spending all its time. My guess would be it's the matrix exponential, which is difficult to speed up. We could try gradient-based optimization here, though, after handling a custom gradient for the step number parameters.

Reply:
Thank you. Will do.

Reply:
Oh, also, I think you only want to do `@threads` in one place. Maybe try only having it over the concentration loop... there might also be a way to write these two loops as one.